### ISSUE TITLE
Null reference on plugin sync updating post image

### ISSUE THREAD

**Issue by TomMalow** [delegateas/Daxif#19]
I ran into a problem when running plugin sync and syncronization attempts to update a post image. 
The image was updated to have fewer attributes.
I got the following trace log:

```
(2018-05-02 10:41:34Z) [Info]: Updating sdkmessageprocessingstepimage: {Plugin Class}: Synchronous Post Create of {Entity}, PostImage
System.Exception: 02/05/2018 08.41.35: An unexpected error occurred.
Errorcode: -2147220970
Trace:


	[======INNER FAULT======]
02/05/2018 08.41.35: System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Crm.ObjectModel.SdkMessageProcessingStepImageServiceInternal`1.Update(IBusinessEntity entity, ExecutionContext context): Microsoft Dynamics CRM has experienced an error. Reference number for administrators or support: #A4494E00
Errorcode: -2147220970
Trace:

   at DG.Daxif.Common.CrmUtility.raiseExceptionIfFault(OrganizationServiceFault fault)
   at DG.Daxif.Modules.Plugin.MainHelper.update@79-2.Invoke(OrganizationServiceFault fault)
   at DG.Daxif.Common.CrmDataHelper.func2@1[a](FSharpFunc`2 faultHandler, FSharpFunc`2 resultTransform, ExecuteMultipleResponseItem[] array)
   at DG.Daxif.Common.CrmDataHelper.performAsBulkResultHandling@96.Invoke(OrganizationRequest[] x)
   at DG.Daxif.Modules.Plugin.MainHelper.update(OrganizationServiceProxy proxy, MapDiff`3 imgDiff, MapDiff`3 stepDiff)
   at DG.Daxif.Modules.Plugin.MainHelper.performSync[a](OrganizationServiceProxy proxy, a solutionName, AssemlyLocal asmCtx, FSharpOption`1 asmReg, FSharpMap`2 sourceTypes, FSharpMap`2 sourceSteps, FSharpMap`2 sourceImgs, FSharpMap`2 targetTypes, FSharpMap`2 targetSteps, FSharpMap`2 targetImgs)
   at DG.Daxif.Modules.Plugin.Main.syncSolution[a](FSharpFunc`2 proxyGen, String projectPath, String dllPath, String solutionName, PluginIsolationMode isolationMode, Boolean ignoreOutdatedAssembly, Boolean dryRun)
   at DG.Daxif.Plugin.Sync(Environment env, String assemblyPath, String projectPath, String solutionName, FSharpOption`1 dryRun, FSharpOption`1 isolationMode, FSharpOption`1 ignoreOutdatedAssembly, FSharpOption`1 logLevel)
   at <StartupCode$FSI_0003>.$FSI_0003.main@() in C:\Users\tkm\Documents\Repos\Bunker Holding CRM\DG.BunkerHolding\DG.BunkerHolding.SalesManagement\Scripts\Daxif\branch\AP\PluginSyncDev.fsx:line 11
Stopped due to error
```

**Comment by TomMalow** [comment]
Workaround it so either delete the image or the plugin itself an run synchronization again. (However, this might lead to duplicate plugins registration after deployment in target environment)

**Comment by TomMalow** [comment]
Fix released in 4.2.10
